app/contracts/new_representation_order_contract.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+class NewRepresentationOrderContract < ApplicationContract
+  option :email_validator, default: -> { CommonPlatform::EmailValidator }
+  option :phone_validator, default: -> { CommonPlatform::PhoneValidator }
+  option :postcode_validator, default: -> { CommonPlatform::PostcodeValidator }
+  config.validate_keys = true
+
+  STATUS_CODES = %w[AP FB FJ FM WD G2 GQ GR].freeze
+
+  json do
+    required(:maat_reference).value(:integer)
+    required(:status_code).value(:string)
+    required(:effective_start_date).value(:date)
+    optional(:effective_end_date).value(:date)
+    required(:defence_organisation).hash do
+      required(:organisation).hash do
+        required(:name).filled(:string)
+        optional(:address).hash do
+          required(:address1).filled(:string)
+          optional(:address2).filled(:string)
+          optional(:address3).filled(:string)
+          optional(:address4).filled(:string)
+          optional(:address5).filled(:string)
+          optional(:postcode).filled(:string)
+        end
+        optional(:contact).hash do
+          optional(:home).filled(:string)
+          optional(:work).filled(:string)
+          optional(:mobile).filled(:string)
+          optional(:primary_email).filled(:string)
+          optional(:secondary_email).filled(:string)
+          optional(:fax).filled(:string)
+        end
+      end
+      required(:laa_contract_number).filled(:string)
+      optional(:sra_number).filled(:string)
+      optional(:bar_council_membership_number).filled(:string)
+      optional(:incorporation_number).filled(:string)
+      optional(:registered_charity_number).filled(:string)
+    end
+    required(:offence_id).value(:string)
+  end
+
+  rule(:offence_id).validate(:uuid)
+
+  rule(:maat_reference) do
+    key.failure('must not exceed 9 digits') if key? && value > 999_999_999
+  end
+
+  rule('defence_organisation.organisation.address.postcode') do
+    key.failure('is not a valid postcode') if key? && !postcode_validator.call(postcode: value)
+  end
+
+  rule('defence_organisation.organisation.contact.primary_email') do
+    key.failure('is not a valid email') if key? && !email_validator.call(email: value)
```

Comment:
**not a blocker**
```suggestion
    key.failure('is not a valid email') if [key?, !email_validator.call(email: value)].all?
```
_not convinced this is clearer/simpler but just putting it out there!_

Reply:
Added this, seems cleaner :+1: